src/main/scala/cromiam/sam/SamClient.scala

```diff
@@ -1,26 +1,65 @@
 package cromiam.sam
 
-import akka.actor.{ActorRef, ActorRefFactory}
-import cromiam.sam.SamActor.{AuthorizationRequest, RegisterWorkflow, RequestAuthorization}
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.headers.Authorization
+import akka.http.scaladsl.model._
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
 
-import scala.concurrent.{ExecutionContext, Future, Promise}
-import cats.implicits._
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+import cats.syntax.functor._
+import cats.syntax.traverse._
+import cats.instances.future._
+import cats.instances.list._
+import cromiam.sam.SamClient._
+import cromiam.server.config.CromIamServerConfig
 
 trait SamClient {
-  protected def samActor: ActorRef
-  protected def actorRefFactory: ActorRefFactory
 
-  protected def requestAuth(authorizationRequest: AuthorizationRequest): Future[Unit] = {
-    val completeMe: Promise[Unit] = Promise()
-    samActor.tell(RequestAuthorization(authorizationRequest, completeMe), ActorRef.noSender)
-    completeMe.future
+  implicit def system: ActorSystem
+  implicit def executor: ExecutionContextExecutor
+  implicit def materializer: ActorMaterializer
+
+  protected def configuration: CromIamServerConfig
+
+  protected def requestAuth(authorizationRequest: WorkflowAuthorizationRequest): Future[Unit] = {
+    def validateEntityBytes(byteString: ByteString): Future[Unit] = if (byteString.utf8String == "true") Future.successful(()) else Future.failed(SamDenialException)
+    val request = samAuthorizationRequest(authorizationRequest)
+    for {
+      response <- Http().singleRequest(request)
+      entityBytes <- response.entity.dataBytes.runFold(ByteString(""))(_ ++ _)
+      _ <- validateEntityBytes(entityBytes)
+    } yield ()
   }
 
-  protected def registerCreation(user: String, workflowIds: List[String])(implicit ec: ExecutionContext): Future[Unit] = {
-    (workflowIds traverse { workflowId =>
-      val completeMe: Promise[Unit] = Promise()
-      samActor.tell(RegisterWorkflow(user, workflowId, completeMe), ActorRef.noSender)
-      completeMe.future
-    }).void
+  protected def registerCreation(authorization: Authorization, workflowIds: List[String])(implicit ec: ExecutionContext): Future[Unit] = {
+    def registerId(id: String) = Http().singleRequest(samRegisterPost(authorization, id)) map { _.discardEntityBytes() }
```

Comment:
If the response is anything other than a `2xx` then the calling method in `CromIamApiService` catches it and returns an error like "Unable to connect to Sam during during new workflow creation".